itk_dev_shared_components/kmd_nova/nova_notes.py

```diff
@@ -0,0 +1,139 @@
+"""This module has functions to do with journal note related calls to the KMD Nova api."""
+
+import base64
+import uuid
+import urllib.parse
+from datetime import datetime
+
+import requests
+
+from itk_dev_shared_components.kmd_nova.authentication import NovaAccess
+from itk_dev_shared_components.kmd_nova.nova_objects import JournalNote
+
+
+def add_text_note(case_uuid: str, note_title: str, note_text: str, nova_access: NovaAccess) -> str:
+    """Add a text based journal note to a Nova case.
+
+    Args:
+        case_uuid: The uuid of the case to add the journal note to.
+        note_title: The title of the note.
+        note_text: The text content of the note.
+        nova_access: The NovaAccess object used to authenticate.
+
+    Returns:
+        The uuid of the created journal note.
+    """
+    note_uuid = str(uuid.uuid4())
+
+    url = urllib.parse.urljoin(nova_access.domain, "api/Case/Update")
+    params = {"api-version": "1.0-Case"}
+
+    payload = {
+        "common": {
+            "transactionId": str(uuid.uuid4()),
+            "uuid": case_uuid
+        },
+        "journalNotes": [
+            {
+                "uuid": note_uuid,
+                "approved": False,
+                "journalNoteAttributes": {
+                    "journalNoteDate": datetime.today().isoformat(),
+                    "title": note_title,
+                    "journalNoteType": "Bruger",
+                    "format": "Text",
+                    "note": _encode_text(note_text)
+                }
+            }
+        ]
+    }
+
+    headers = {'Content-Type': 'application/json', 'Authorization': f"Bearer {nova_access.get_bearer_token()}"}
+
+    response = requests.patch(url, params=params, headers=headers, json=payload, timeout=60)
+    response.raise_for_status()
+
+    return note_uuid
+
+
+def _encode_text(string: str) -> str:
+    """Encode a string to a base64 string.
+    Ensure the base64 string doesn't contain padding by
+    inserting spaces at the end of the input string.
```

Comment:
You should add a comment on why you're doing this. Otherwise, people (and maybe even your future self) may think you've lost your marbles …

Furthermore, it may be worth noting that the added spaces apparently and magically disappear in KMD nova.